comp/core/settings/settingsimpl/settingsimpl_test.go

```diff
@@ -120,6 +120,23 @@ func TestRuntimeSettings(t *testing.T) {
 				assert.NotEqual(t, "", string(body))
 			},
 		},
+		{
+			"GetFullConfigWithoutDefaults",
+			func(t *testing.T, comp settings.Component) {
+				responseRecorder := httptest.NewRecorder()
+				request := httptest.NewRequest("GET", "http://agent.host/test/", nil)
+
+				comp.GetFullConfigWithoutDefaults("")(responseRecorder, request)
+				resp := responseRecorder.Result()
+				defer resp.Body.Close()
+				body, _ := io.ReadAll(resp.Body)
+
+				assert.Equal(t, 200, responseRecorder.Code)
+				// The full config is too big to assert against
+				// Ensure the response body is not empty to validate we wrote something
+				assert.NotEqual(t, "", string(body))
```

Comment:
You should set at least 1 setting as a default using a config mock and assert that that setting is not there. Right now you're just testing that something was returned not that it's valid.